providers/oidc.go

```diff
@@ -279,27 +284,32 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 	return claims, nil
 }
 
-func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {
+func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) ([]string, error) {
 	groups := []string{}
 
 	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
 	if rawGroups != nil && ok {
 		for _, rawGroup := range rawGroups {
 			group, ok := rawGroup.(string)
-			if ok {
-				groups = append(groups, group)
+			if !ok {
```

Comment:
Do you think it is worth getting this string vs object logic in a helper method? In case the other corner cases around groups claims (E.g. string vs array claim in the other PR) also need to utilize this logic?

Reply:
What alternative are you suggesting? :)

Reply:
Everything from lines 293-301 in the `for` loop as a helper method.  Whatever logic you have here for the array case, the string case in the other PR will need to utilize. Let's make it a helper method.